### PR TITLE
Restriction out-of-bounds checks

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
Adds checks for restrictions accessing inputs/outputs that are out of bounds. Prevents panics at runtime.

Also adds 3 new restrictions so we can cover original-token-id behaviour in full

Also adds renovate config at @dblane-digicatapult's request

### CODE REVIEW

File: pallets/process-validation/src/restrictions.rs [hunk 1]
```diff
@@ -26,6 +26,18 @@ pub enum Restriction<RoleKey, TokenMetadataKey, TokenMetadataValue, TokenMetadat
...
     },
+    OutputHasMetadataValue {
+        index: u32,
+        metadata_key: TokenMetadataKey
+    },
+    InputHasRole {
+        index: u32,
+        role_key: RoleKey
+    },
+    InputHasMetadataValue {
```
Comment (jonmattgray) [inline pallets/process-validation/src/restrictions.rs:37]: should `HasMetadataValue` be `HasMetadataKey`?

Comment (mattdean-digicatapult) [reply pallets/process-validation/src/restrictions.rs:37]: 🤔 , maybe `HasMetadata`? Because it's whether it has that metadata entry

Comment (jonmattgray) [reply pallets/process-validation/src/restrictions.rs:37]: yeah `HasMetadata` would match `HasRole`

Comment (jonmattgray) [reply pallets/process-validation/src/restrictions.rs:37]: in that they have a similar name and are both checking there is a role/metadata entry with any value

File: pallets/process-validation/src/restrictions.rs [hunk 2]
```diff
@@ -26,6 +26,18 @@ pub enum Restriction<RoleKey, TokenMetadataKey, TokenMetadataValue, TokenMetadat
         index: u32,
         role_key: RoleKey
     },
+    OutputHasMetadataValue {
```
Comment (jonmattgray) [inline pallets/process-validation/src/restrictions.rs:29]: new restrictions can be added to the README

**Review by jonmattgray — APPROVED** [review]

**Pull request merged**